File: pkg/codegen/testing/test/testdata/output-funcs-aws-pp/dotnet/output-funcs-aws.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    var aws_vpc = new Aws.Ec2.Vpc("aws_vpc", new()
    {
        CidrBlock = "10.0.0.0/16",
        InstanceTenancy = "default",
    });

    var privateS3VpcEndpoint = new Aws.Ec2.VpcEndpoint("privateS3VpcEndpoint", new()
    {
        VpcId = aws_vpc.Id,
        ServiceName = "com.amazonaws.us-west-2.s3",
    });

    var privateS3PrefixList = Aws.Ec2.GetPrefixList.Invoke(new()
    {
        PrefixListId = privateS3VpcEndpoint.PrefixListId,
    });

    var bar = new Aws.Ec2.NetworkAcl("bar", new()
    {
        VpcId = aws_vpc.Id,
    });

    var privateS3NetworkAclRule = new Aws.Ec2.NetworkAclRule("privateS3NetworkAclRule", new()
    {
        NetworkAclId = bar.Id,
        RuleNumber = 200,
        Egress = false,
        Protocol = "tcp",
        RuleAction = "allow",
        CidrBlock = privateS3PrefixList.Apply(getPrefixListResult => getPrefixListResult.CidrBlocks[0]),
        FromPort = 443,
        ToPort = 443,
    });

    // A contrived example to test that helper nested records ( `filters`
    // below) generate correctly when using output-versioned function
    // invoke forms.
    var amis = Aws.Ec2.GetAmiIds.Invoke(new()
    {
        Owners = new[]
        {
            bar.Id,
        },
        Filters = new[]
        {
            new Aws.Ec2.Inputs.GetAmiIdsFilterInputArgs
            {
                Name = bar.Id,
                Values = new[]
                {
                    "pulumi*",
                },
            },
        },
    });

});

File: tests/testdata/codegen/output-funcs-aws-pp/go/output-funcs-aws.go
package main

import (
	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ec2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		aws_vpc, err := ec2.NewVpc(ctx, "aws_vpc", &ec2.VpcArgs{
			CidrBlock:       "10.0.0.0/16",
			InstanceTenancy: "default",
		})
		if err != nil {
			return err
		}
		privateS3VpcEndpoint, err := ec2.NewVpcEndpoint(ctx, "privateS3VpcEndpoint", &ec2.VpcEndpointArgs{
			VpcId:       aws_vpc.ID(),
			ServiceName: pulumi.String("com.amazonaws.us-west-2.s3"),
		})
		if err != nil {
			return err
		}
		privateS3PrefixList := ec2.GetPrefixListOutput(ctx, ec2.GetPrefixListOutputArgs{
			PrefixListId: privateS3VpcEndpoint.PrefixListId,
		}, nil)
		bar, err := ec2.NewNetworkAcl(ctx, "bar", &ec2.NetworkAclArgs{
			VpcId: aws_vpc.ID(),
		})
		if err != nil {
			return err
		}
		_, err = ec2.NewNetworkAclRule(ctx, "privateS3NetworkAclRule", &ec2.NetworkAclRuleArgs{
			NetworkAclId: bar.ID(),
			RuleNumber:   pulumi.Int(200),
			Egress:       false,
			Protocol:     pulumi.String("tcp"),
			RuleAction:   pulumi.String("allow"),
			CidrBlock: privateS3PrefixList.ApplyT(func(privateS3PrefixList ec2.GetPrefixListResult) (string, error) {
				return privateS3PrefixList.CidrBlocks[0], nil
			}).(pulumi.StringOutput),
			FromPort: 443,
			ToPort:   443,
		})
		if err != nil {
			return err
		}
		// A contrived example to test that helper nested records ( `filters`
		// below) generate correctly when using output-versioned function
		// invoke forms.
		_ = ec2.GetAmiIdsOutput(ctx, ec2.GetAmiIdsOutputArgs{
			Owners: pulumi.StringArray{
				bar.ID(),
			},
			Filters: []ec2.GetAmiIdsFilterArgs{
				{
					Name: bar.ID(),
					Values: pulumi.StringArray{
						pulumi.String("pulumi*"),
					},
				},
			},
		}, nil)
		return nil
	})
}
